quarkus-picocli/src/test/java/PicocliIT.java

```diff
@@ -0,0 +1,118 @@
+import static java.util.concurrent.CompletableFuture.runAsync;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+
+@QuarkusScenario
+public class PicocliIT {
+
+    static final String GREETING_COMMAND_DESC = "greeting";
+    static final String GREETING_NAME_OPTION = "--name";
+    static final String GREETING_NAME_VALUE = "QE";
+
+    static final String AGE_COMMAND_DESC = "age";
+    static final String AGE_OPTION = "--age";
+    static final String AGE_VALUE = "30";
+
+    @QuarkusApplication
+    static final RestService greetingApp = new RestService()
+            .withProperty("quarkus.args",
+                    GREETING_COMMAND_DESC + " " + GREETING_NAME_OPTION + " " + GREETING_NAME_VALUE)
+            .withProperty("quarkus.profile", "dev")
+            .setAutoStart(false);
+
+    @QuarkusApplication
+    static final RestService ageApp = new RestService()
+            .withProperty("quarkus.args",
+                    AGE_COMMAND_DESC + " " + AGE_OPTION + " " + AGE_VALUE)
+            .withProperty("quarkus.profile", "dev")
+            .setAutoStart(false);
+
+    @QuarkusApplication
+    static final RestService greetingBlankArgumentApp = new RestService()
+            .withProperty("quarkus.args",
+                    "" + " " + GREETING_NAME_OPTION + " " + GREETING_NAME_VALUE)
+            .withProperty("quarkus.profile", "dev")
+            .setAutoStart(false);
+
+    @QuarkusApplication
+    static final RestService greetingInvalidArgumentApp = new RestService()
+            .withProperty("quarkus.args",
+                    GREETING_COMMAND_DESC + " " + "-x" + " " + GREETING_NAME_VALUE)
+            .withProperty("quarkus.profile", "dev")
+            .setAutoStart(false);
+
+    @QuarkusApplication
+    static final RestService bothTopCommandApp = new RestService()
+            .withProperty("quarkus.profile", "dev")
+            .setAutoStart(false);
+
+    @Test
+    public void verifyErrorForApplicationScopedBeanInPicocliCommand() {
+        String expectedAgeOutput = String.format("Your age is: %s", AGE_VALUE);
+        try {
+            runAsync(ageApp::start);
+            ageApp.logs().assertContains(expectedAgeOutput);
+        } finally {
+            ageApp.stop();
+        }
+    }
+
+    @Test
+    public void verifyGreetingCommandOutputsExpectedMessage() throws InterruptedException {
+        String expectedOutput = String.format("Hello %s!", GREETING_NAME_VALUE);
+        try {
+            runAsync(greetingApp::start);
+            greetingApp.logs().assertContains(expectedOutput);
+            Thread.sleep(3500);
```

Comment:
Why wait?

Reply:
It was for local testing purposes, I forgot clean it, 'll remove it, thanks